src/core/network.js

```diff
@@ -414,9 +415,27 @@ PDFNetworkStreamFullRequestReader.prototype = {
       networkManager.abortRequest(fullRequestXhrId);
     }
 
+    // Content-Disposition: attachment; filename=Naïve file.txt
+    var fullRequestXhr = networkManager.getRequestXhr(fullRequestXhrId);
+    var contentDisposition =
+      fullRequestXhr.getResponseHeader('Content-Disposition');
+    this._fileName = this._extractNameFrom(contentDisposition);
+
     this._headersReceivedCapability.resolve();
   },
 
+  _extractNameFrom:
+      function PDFNetworkStreamFullRequestReader_extractNameFrom(cDisposition) {
+    if (cDisposition) {
+      var parts = /.+; filename="(.+)"/gi.exec(cDisposition);
```

Comment:
This is a good point.

The patch is pretty simple. It parses the content disposition header in network layer and sends up to the viewer UI the result. The UI uses it to compose the title and the download file name.

Adding some kind of validation that prevents the viewer from loading invalid files and making it fail in a facy way would be certainly an improvement.

I can do it, but obviously it will take some extra  time since I'm not familiar with the procedures that you follow to raise exceptions  from the network layer to the UI. 

At this point I would suggest to split the work in two. First finish integrating this pull request in the main branch with the current functionality. It will solve a couple of issues and add a feature long awaited. 

Then we can create an issue to add the feature that you suggest. I'm not member of the official project, and I don't want to write permissions to the main repo. I'm a newbie that will need some time to have a general vision of all the project, but, if it's feasible, you can asssign the issue to me and I will work on it in a new branch after this one is acceped and merged.

In case you consider this pull request should not be merged without this feature it would be nice if you could give me some advice about how to handle it. A file name with a line number would be enough: "Hi Juan, look at xxxx.js line yyy. There we do something similar to what you need to achieve".

Kind regards.